### DISCUSSION TITLE
why eplace cannot optimize Innovus's result anymore

### DISCUSSION THREAD

**Discussion by tigereatsheep** [The-OpenROAD-Project/OpenROAD#4829] [General]
I just implemented a placement tool(eplace based, like openroad), and I use this tool to optimize innovus's placement result, it just stalled, so wired

**Comment by maliberty** [comment]
Perhaps there is no further improvement to be had?  I'm not sure how this relates to OR.

**Reply by tigereatsheep** [reply]
Thank you for you reply. Maybe you're right. I just compared DCG/INVS/OR 's placement result carefully (a complex case), found that DCG and INVS's results were low congested, but OR's congestion is abnormally high. I think it's local congestion, do you know how to solve this?

**Reply by maliberty** [reply]
I can't comment on other tools or their results.  Are you using -routability in OR?

**Reply by tigereatsheep** [reply]
No, but I've read the source code, -routability only change the overflow/targetdensity parameters, this do little help to solve congestion in my own "eplace based" tools.

**Reply by maliberty** [reply]
It bloats the cells which spreads them out more in the congested areas.

**Reply by tigereatsheep** [reply]
Hi Matt, Thank you very much. OR solved the congestion problem, the core area util of design is 68%, I just set the target density from 1.0 to 0.71 and the congestion rate down from 15% to 2%. OR is really amazing project

**Reply by maliberty** [reply]
I'm glad you found a solution.
